internal/build/phase_test.go

```diff
@@ -446,3 +475,62 @@ func CreateFakeLifecycleExecution(logger logging.Logger, docker client.CommonAPI
 		NoProxy:    "some-no-proxy",
 	})
 }
+
+// helper function to expose standard UNIX socket `/var/run/docker.sock` via TCP localhost:12345
+func forwardUnix2TCP(ctx context.Context, t *testing.T) {
+	wg := sync.WaitGroup{}
+
+	forwardCon := func(tcpCon net.Conn) {
+		defer wg.Done()
+
+		defer tcpCon.Close()
+		go func() {
+			<-ctx.Done()
+			tcpCon.Close()
+		}()
+
+		unixCon, err := net.Dial("unix", "/var/run/docker.sock")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer unixCon.Close()
+		go func() {
+			<-ctx.Done()
+			unixCon.Close()
+		}()
+
+		buffA := make([]byte, 4096)
+		buffB := make([]byte, 4096)
+		go io.CopyBuffer(tcpCon, unixCon, buffA)
+		io.CopyBuffer(unixCon, tcpCon, buffB)
+	}
+
+	listener, err := net.Listen("tcp4", ":12345")
```

Comment:
@dfreilich shall I randomize port or can we live with having it hardcoded?

Reply:
Hmmm. Hardcoded should work for now, though I worry that it could somehow cause a collision with another test, though we don't really listen on any other ports. Can you pull the port out and make it an explicit variable in the test file though, to make it clearer?